pkg/catalog/catalog.go

```diff
@@ -2260,7 +2276,23 @@ func (c *Catalog) runBackgroundTaskSteps(repository *graveler.RepositoryRecord,
 			if err != nil {
 				log.WithError(err).WithField("step", step.Name).Errorf("Catalog background task step failed")
 				task.Done = true
-				task.Error = err.Error()
+				// Classify the error using the API callback (handleAPIErrorCallback from the controller)
+				// before the original error is lost when stored in protobuf, and populate the task's error details.
+				c.APIErrorCB(ctx, nil, nil, err, func(w http.ResponseWriter, r *http.Request, code int, v interface{}) {
+					if code >= 0 && code <= 999 {
+						task.StatusCode = int32(code)
+					} else {
+						task.StatusCode = 500
+					}
+					switch e := v.(type) {
+					case string:
+						task.ErrorMsg = e
+					case error:
+						task.ErrorMsg = fmt.Sprint(err)
+					default:
+						task.ErrorMsg = fmt.Sprintf("%v", v)
+					}
```

Comment:
If `v != nil` -> it can simply be  `fmt.Sprintf("%v", v)` for all cases (error, string, any)

Reply:
Asked for re-review but I don't see change in here? 
all 3 cases, type `string`, type `error` and else can be simply `fmt.Sprintf("%v", v)`